Mathlib/Algebra/MonoidAlgebra/NoZeroDivisors.lean

```diff
@@ -72,15 +72,14 @@ theorem mul_apply_mul_eq_mul_of_uniqueMul [Mul A] {f g : MonoidAlgebra R A} {a0
   classical
   simp_rw [mul_apply, sum, ← Finset.sum_product']
   refine (Finset.sum_eq_single (a0, b0) ?_ ?_).trans (if_pos rfl) <;> simp_rw [Finset.mem_product]
-  · refine fun ab hab hne => if_neg (fun he => hne <| Prod.ext ?_ ?_)
+  · refine fun ab hab hne ↦ if_neg (fun he ↦ hne <| Prod.ext ?_ ?_)
     exacts [(h hab.1 hab.2 he).1, (h hab.1 hab.2 he).2]
-  · refine fun hnotMem => ite_eq_right_iff.mpr (fun _ => ?_)
+  · refine fun hnotMem ↦ ite_eq_right_iff.mpr (fun _ ↦ ?_)
     rcases not_and_or.mp hnotMem with af | bg
     · rw [notMem_support_iff.mp af, zero_mul]
     · rw [notMem_support_iff.mp bg, mul_zero]
 
-instance instNoZeroDivisorsOfUniqueProds [NoZeroDivisors R] [Mul A] [UniqueProds A] :
-    NoZeroDivisors (MonoidAlgebra R A) where
+instance [NoZeroDivisors R] [Mul A] [UniqueProds A] : NoZeroDivisors (MonoidAlgebra R A) where
```

Comment:
The original author might have chosen to name this because these instances are invoked by name in proofs, but this is no longer necessary with `inferInstanceAs`. Also the name is the same as the current auto-generated name, so I've removed the name here as well.